services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/ThrottlingRerouteSpec.groovy

```diff
@@ -294,6 +294,37 @@ class ThrottlingRerouteSpec extends BaseSpecification {
         db.countFlows() == 0
     }
 
+    def "Auto-reroute on the flow which is already deleted should not revive the flow"() {
+        given: "A flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+        def path = PathHelper.convert(northboundService.getFlowPath(flow.id))
+
+        when: "Remove the flow"
+        northboundService.deleteFlow(flow.id)
+
+        and: "Immediately break the path to init all flows on that path to reroute"
+        def islToBreak = pathHelper.getInvolvedIsls(path).first()
```

Comment:
IMHO, would be better to evaluate this before deletion of the flow to conserve some time.